CRM/Contact/Import/Parser/Contact.php

```diff
@@ -2244,24 +2218,6 @@ protected function handleDuplicateError(array $newContact, $statusFieldName, arr
   public function deprecated_validate_formatted_contact(&$params): void {
     // Look for offending email addresses
 
-    if (array_key_exists('email', $params)) {
-      foreach ($params['email'] as $count => $values) {
-        if (!is_array($values)) {
-          continue;
-        }
-        if ($email = CRM_Utils_Array::value('email', $values)) {
-          // validate each email
-          if (!CRM_Utils_Rule::email($email)) {
-            throw new CRM_Core_Exception('No valid email address');
-          }
-
-          // check for loc type id.
-          if (empty($values['location_type_id'])) {
```

Comment:
I don't think it can really get to this point with no location_type_id as that is added during mapping - this is old code that used to be in apiv2.